astroquery/imcce/core.py

```diff
@@ -0,0 +1,644 @@
+        | ``hourangle``    | Target hour angle (deg, float, 4, 5)          |
+        +------------------+-----------------------------------------------+
+        | ``siderealtime`` | Local sidereal time (hr, float, 5, 6)         |
+        +------------------+-----------------------------------------------+
+        | ``refraction``   | Atmospheric refraction (arcsec, float, 5, 6)  |
+        +------------------+-----------------------------------------------+
+        | ``airmass``      | Target airmass (float, 5, 6)                  |
+        +------------------+-----------------------------------------------+
+        | ``posunc``       | Positional uncertainty (arcsec, float, 5, 6)  |
+        +------------------+-----------------------------------------------+
+
+
+        Examples
+        --------
+
+        >>> from astroquery.imcce import Miriade
+        >>> from astropy.time import Time
+        >>> epoch = Time('2019-01-01', format='iso')
+        >>> Miriade.get_ephemerides('3552', epoch=epoch)  # doctest: +SKIP
+        <Table masked=True length=1>
+           target          epoch                 RA         ...  DEC_rate   delta_rate
+                             d                  deg         ... arcs / min    km / s
+          bytes20         float64             float64       ...  float64     float64
+        ----------- -------------------- ------------------ ... ---------- ------------
+        Don Quixote            2458484.5 16.105294999999998 ...   -0.25244   31.4752734
+        """
+
+        URL = conf.ephemcc_server
+        TIMEOUT = conf.timeout
+
+        if isinstance(epoch, (int, float)):
+            epoch = Time(epoch, format='jd')
+        elif isinstance(epoch, str):
+            epoch = Time(epoch, format='iso')
+        elif epoch is None:
+            epoch = Time.now()
+
+        request_payload = OrderedDict([
+            ('-name', targetname),
+            ('-type', objtype[0].upper()+objtype[1:]),
+            ('-ep', str(epoch.jd)),
+            ('-step', epoch_step),
+            ('-nbd', epoch_nsteps),
+            ('-observer', location),
+            ('-output', '--jul'),
+            ('-tscale', timescale),
+            ('-theory', planetary_theory),
+            ('-teph', ephtype),
+            ('-tcoor', coordtype),
+            ('-rplane', {'equator': 1, 'ecliptic': 2}[refplane]),
+            ('-oscelem', elements),
+            ('-mime', 'votable')])
+
+        if radial_velocity:
+            request_payload['-output'] += ',--rv'
+
+        if get_query_payload:
+            return request_payload
+
+        # query and parse
+        response = self._request('GET', URL, params=request_payload,
+                                 timeout=TIMEOUT, cache=cache)
+        self._query_uri = response.url
+
+        if get_raw_response:
+            self._get_raw_response = True
+        else:
+            self._get_raw_response = False
```

Comment:
this stateful approach to getting the raw data is a little awkward; could you comment briefly (either in the code or the PR) on why it's necessary?

Reply:
Yes, this is indeed awkward and I have no clue why I did it this way... I changed the approach and now use a class-internal variable. If there is a better approach, please let me know! Commit coming soon...